### ISSUE TITLE
Bundle `@types/node`

### ISSUE THREAD

**Issue by lishaduck** [denoland/deno#27046]
Deno has a Node compatibility layer. I'm currently using it for a few operations in `@lishaduck/effect-platform-deno` that rely on fids. However, the `node:*` modules don't come with types. The docs [recommend](https://docs.deno.com/runtime/fundamentals/node/#including-node-types) installing `@types/node`, but this both augments the globals (e.g., it shows `Buffer` as available) and isn't kept in sync with Deno's TS and Node support. For example, until #26950 is completed, the latest `@types/node` doesn't work with Deno, and it'd also be nice to get deprecation warnings on things that the Node-compat layer doesn't support.

**Comment by VienDinhCom** [comment]
Maybe we should wait for Node to support TypeScript, so we can have more stable typing.

https://nodejs.org/en/learn/typescript/run-natively

**Comment by lishaduck** [comment]
> Maybe we should wait for Node to support TypeScript, so we can have more stable typing.

I haven't seen anything about Node taking over maintenance of `@types/node` as part of the ts work.
That'd be great, but I don't foresee it happening anytime soon—I believe Node uses a custom JSDoc-like comment-based dsl for documenting types.

**Comment by bombillazo** [comment]
Hey, we're having an issue related to Buffer after upgrading to v2, and Im sure its related to how the lib handles Node identifiers and types. Any solution to this?

**Comment by lishaduck** [comment]
Yeah, you need to downgrade `@types/node` to `<22.7.3`, versions higher than that require TS 5.7 (tracked by #26950).

**Comment by lishaduck** [comment]
I believe this was implemented ~~in Deno 2.4: ["Additionally, Deno 2.4 will use @types/node version 22.15.14 by default when type checking."](https://deno.com/blog/v2.4#nodejs-api-support:~:text=Additionally%2C%20Deno%202.4%20will%20use%20%40types/node%20version%2022.15.14%20by%20default%20when%20type%20checking.)~~ forever ago?
